recipes-openxt/openxt/openxt-measuredlaunch/ml-functions

```diff
@@ -376,12 +415,38 @@ generate_policy() {
         lcp_writepol -i $tboot_idx -f $policy -p "$password" 2>&1 1>/dev/null
     fi
 
-    if [ $? -ne 0 ]; then
+    if [[ $? -ne 0 ]]; then
         echo "failed to write tboot Verified Launch policy to TPM" >&2
         return 2
     fi
 
-    [ -f $policy ] && rm -f $policy
+    [[ -f $policy ]] && rm -f $policy
+}
+
+# Function to create policy and store into a file
+# parameter 0: path where policy file should be written
+# parameter 1: hash algorithm to use, either sha1 or sha256
+# return 0 on success
+# return 1 if fails to read
+create_policy() {
+    local policy=$1
+    local hash_alg=${2:-"sha1"}
+
+    if [[ "${hash_alg}" != "sha1" ]] && [[ "${hash_alg}" != "sha256" ]]; then
+        echo "invalid algorithm" && return 1
+    fi
+
+    [[ -f $policy ]] && rm -f $policy
+
+    tb_polgen --create --type nonfatal --ctrl 1 --alg $hash_alg $policy 2>&1 1>/dev/null && \
```

Comment:
This will discard stdout and replace it with stderr, is this what you want?
`1>/dev/null 2>&1` would discard everything.
(more occurrences below)